xdsl/ir.py

```diff
@@ -544,7 +544,8 @@ def build(
         | None = None,
         attributes: dict[str, Attribute] | None = None,
         successors: Sequence[Block] | None = None,
-        regions: Sequence[Region | Sequence[Operation] | Sequence[Block]]
+        regions: Sequence[Region | Sequence[Operation] | Sequence[Block]
+                          | Sequence[Region]]
```

Comment:
First Op with `OptRegion` 🍾  This actually shows some more typing issues we had, where the function always accepted lists of regions and lists of lists of blocks etc but the type wasn't as expressive. I have another PR coming that makes it all a bit clearer, but just added the types to the function argument for now